p2p/registration.go

```diff
@@ -43,6 +42,8 @@ func NewRegistration(expireTime time.Duration) PeerRegisteration {
 
 // RegisteredPeers return current registered peers' ID
 func (r *BasicPeerRegistration) RegisteredPeers() []peerlib.ID {
+	r.RLock()
```

Comment:
Should this one r.Lock() ? because it writes the item

Reply:
there is no write to "r"